frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -218,38 +225,54 @@ const VariableDropdown: React.FC<VariableDropdownProps> = ({ id, name, namespace
 
   const [isError, setIsError] = React.useState(false);
 
+  const dataSourceName = variable?.datasource?.name;
+  const extensions = useExtensions<DataSourceExtension>(isDataSource);
+  const extension = extensions.find((ext) => ext.pluginName === 'dashboards-datasource-plugin');
+
+  const getURL = React.useCallback(
+    async (prometheusProps) => {
+      if (!dataSourceName) {
+        return getPrometheusURL(prometheusProps);
+      }
+      const getDataSource = await extension.properties.getDataSource();
```

Comment:
We need to test when the plugin extension does not exists or fails to load, basically validate that `extension` exists. Testing in the cluster, the filtering by `pluginName` returns `undefined` because the plugin name is  `dashboards-datasource-plugin` not `dashboards-datasource-plugin`, the extension was not found and no dashboards were rendered.

Reply:
This code is now wrapped in a try/catch